app/models/question.rb

```diff
@@ -35,4 +39,24 @@ def current_user_downvote_correction(creator_id)
       downvotes.where(creator:creator_id).destroy_all
     end
   end
+
+  def self.show_all_questions(category)
+    if category
+      where(category: Category.find(category))
+    else
+      order_by_update
+    end
+  end
+
+  def question_upvotes
+    upvotes.count
+  end
+
+  def question_downvotes
+    downvotes.count
+  end
+
+  def net_votes
```

Comment:
Agreed with Katy. This is great work. I like how you've moved so much logic down the stack.